src/index.test.ts

```diff
@@ -120,6 +120,35 @@ describe('jsonFetch', function () {
       expect(response.statusText).to.equal('Created');
       expect(response.headers).to.be.ok();
     });
+
+    it('calls onRequestStart when is passed to jsonFetch in JsonFetchOptions', async function () {
+      const onRequestStart = sandbox.stub();
+      nock('http://www.test.com').get('/products/1234').reply(200);
+      await jsonFetch('http://www.test.com/products/1234', {
+        onRequestStart,
+      });
+      expect(onRequestStart.callCount).to.equal(1);
```

Comment:
nitpick: we'll get more useful test output if this fails if we use more expressive assertions
```suggestion
      expect(onRequestStart).to.have.been.calledOnce();
```